network/p2p/blob_service.go

```diff
@@ -45,68 +48,110 @@ func WithReprovideInterval(d time.Duration) network.BlobServiceOption {
 	}
 }
 
+// WithBitswap configures the blobstore to use Bitswap as the underlying exchange
+func WithBitswap(host host.Host, r routing.ContentRouting, prefix string) network.BlobServiceOption {
+	return func(bs network.BlobService) {
+		bs.(*blobService).config.BitswapNetwork = bsnet.NewFromIpfsHost(host, r, bsnet.Prefix(protocol.ID(prefix)))
+		bs.(*blobService).config.ContentRouting = r
+	}
+}
+
+// WithBitswapOptions sets additional options for Bitswap exchange
 func WithBitswapOptions(opts ...bitswap.Option) network.BlobServiceOption {
 	return func(bs network.BlobService) {
 		bs.(*blobService).config.BitswapOptions = opts
 	}
 }
```

Comment:
Let's combine these into a single option

Reply:
I think that would mean we'd have to update the RegisterBlobNetwork to also accept a slice of bitswap options, and pipe that through all of the chained methods. The current implementation creates a pretty clean unified interface.

Reply:
On second thought, I think it would probably be better to create a separate implementation for non-network blob store, because the BlobService API contains methods which don't make sense for local blob store (GetSession, TriggerReprovide).

Reply:
I'll just revert these changes then since the datastore checking functionality was removed.

Reply:
Those methods are currently part of the `BlobService` interface. Searching through our code neither appear to actually be called outside of tests so I can create a new interface that removes them, but I'm not sure if that's going to break anything